sdk/src/Core/Amazon.Runtime/EventStreams/Internal/EventStream.cs

```diff
@@ -253,22 +262,39 @@ protected void Process()
         {
 #if AWS_ASYNC_API
             // Task only exists in framework 4.5 and up, and Standard.
-            Task.Run(() => ProcessLoop());
+            _ = ProcessLoopAsync();
 #else
             // ThreadPool only exists in 3.5 and below. These implementations do not have the Task library.
             ThreadPool.QueueUserWorkItem(ProcessLoop);
 #endif
         }
 
-        /// <summary>
-        /// The background thread main loop. It will constantly read from the network stream until IsProcessing is false, or an error occurs.
-        /// <para></para>
-        /// This stub exists due to FXCop.
-        /// </summary>
-        private void ProcessLoop()
+#if AWS_ASYNC_API
+        private async Task ProcessLoopAsync()
         {
-            ProcessLoop(null);
+            var buffer = new byte[BufferSize];
```

Comment:
How do you feel about optimizing here for memory? Something like
```
#If !NETFRAMEWORK
            var arrayPool = ArrayPool<byte>.Shared;
            var buffer = arrayPool.Rent(BufferSize);
#else
             var buffer = new byte[BufferSize];
#endif
```
Then conditionally adding the using statement based off `!NETFRAMEWORK`

Reply:
FYI I would be very careful with pooling here. The problem is that the byte array is handed out to an event handler that then theoretically can do whatever with the byte array. While this problem already exists today with the current approach at least the byte array is local to the EventStream implementation and therefore "buffered" locally over subsequent reads until no data is available anymore. With the pool you would acquire a buffer from the pool and keep it for a long period and then return the buffer to the pool. If the event handler does something weird that buffer reference would be handed back to the pool and then eventually handed out to someone else

If you want to do pooling it might be better to pool for every read and return after the handler returned or does the decoder do some memory copying under the covers?

Reply:
that's a good point, the decoder does do some memory copying under the covers, but the added complexity of pooling probably isn't worth the benefit here.